app/javascript/src/scenes/SettingsPage/scenes/profile/components/MyProfile.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from "react";
+import { func } from "prop-types";
+import styled from "styled-components";
+import { FormattedMessage } from "react-intl";
+import { getUserProfileInfo } from "../../../../../services/api/users_api";
+
+import AvatarInputField from "./AvatarInputField";
+import ProfileInputField from "./ProfileInputField";
+
+const AvatarLabel = styled.h4`
+  margin-top: 15px;
+  font-size: 13px;
+  font-weight: 700;
+`;
+
+class MyProfile extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      fullName: "",
+      avatarThumb: "",
+      initials: "",
+      email: "",
+      timeZone: "",
+      newEmail: ""
+    };
+    this.loadInfo = this.loadInfo.bind(this)
+  }
+
+  componentDidMount() {
+    this.loadInfo();
+  }
+
+  loadInfo() {
```

Comment:
this is a perfect place to store this kind of stuff into Redux or Mobx store. user is probably loaded only once, right? this will cause fetching user every time this component loads. it's probably loaded all the time but if you accidentially hide it, it will rerender user.